robot/impl/local_robot.go

```diff
@@ -709,12 +723,135 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 	}
 }
 
-// FrameSystemConfig returns the info of each individual part that makes up a robot's frame system.
-func (r *localRobot) FrameSystemConfig(
-	ctx context.Context,
-	additionalTransforms []*referenceframe.LinkInFrame,
-) (framesystemparts.Parts, error) {
-	return r.frameSvc.Config(ctx, additionalTransforms)
+// Config returns the info of each individual part that makes up the frame system
+// The output of this function is to be sent over GRPC to the client, so the client
+// can build its frame system. requests the remote components from the remote's frame system service.
+// NOTE(RDK-258): If remotes can trigger a local robot to reconfigure, you don't need to update remotes in every call.
+func (r *localRobot) FrameSystemConfig(ctx context.Context) (*framesystem.Config, error) {
+	r.fsMu.RLock()
+	defer r.fsMu.RUnlock()
+
+	localParts, err := r.getLocalParts(ctx)
+	if err != nil {
+		return nil, err
+	}
+	remoteParts, err := r.getRemoteParts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &framesystem.Config{Parts: append(localParts, remoteParts...)}, nil
+}
+
+// getLocalParts collects and returns the physical parts of the robot that may have frame info,
+// excluding remote robots and services, etc from the robot's config.Config.
+func (r *localRobot) getLocalParts(ctx context.Context) ([]*referenceframe.FrameSystemPart, error) {
```

Comment:
I don't love the names `getLocalParts` and `getRemoteParts` because they don't have any mention of framesystem-relation. Can we call them `getLocalFrameSystemParts` and `getRemoteFrameSystemParts` or something?

Reply:
Changed